Sources/TokamakCore/StackReconciler.swift

```diff
@@ -201,45 +202,50 @@ public final class StackReconciler<R: Renderer> {
     }.store(in: &mountedApp.persistentSubscriptions)
   }
 
-  private func render<T>(
-    compositeElement: MountedCompositeElement<R>,
-    body bodyKeypath: ReferenceWritableKeyPath<MountedCompositeElement<R>, Any>,
-    result: KeyPath<MountedCompositeElement<R>, (Any) -> T>
-  ) -> T {
+  private func body(
+    of compositeElement: MountedCompositeElement<R>,
+    keyPath: ReferenceWritableKeyPath<MountedCompositeElement<R>, Any>
+  ) -> Any {
     compositeElement.updateEnvironment()
     if let info = typeInfo(of: compositeElement.type) {
       var stateIdx = 0
       let dynamicProps = info.dynamicProperties(
         &compositeElement.environmentValues,
-        source: &compositeElement[keyPath: bodyKeypath]
+        source: &compositeElement[keyPath: keyPath]
       )
 
       compositeElement.transientSubscriptions = []
       for property in dynamicProps {
         // Setup state/subscriptions
         if property.type is ValueStorage.Type {
-          setupStorage(id: stateIdx, for: property, of: compositeElement, body: bodyKeypath)
+          setupStorage(id: stateIdx, for: property, of: compositeElement, body: keyPath)
           stateIdx += 1
         }
         if property.type is ObservedProperty.Type {
-          setupTransientSubscription(for: property, of: compositeElement, body: bodyKeypath)
+          setupTransientSubscription(for: property, of: compositeElement, body: keyPath)
         }
       }
     }
 
-    return compositeElement[keyPath: result](compositeElement[keyPath: bodyKeypath])
+    return compositeElement[keyPath: keyPath]
   }
 
   func render(compositeView: MountedCompositeView<R>) -> AnyView {
-    render(compositeElement: compositeView, body: \.view.view, result: \.view.bodyClosure)
+    let view = body(of: compositeView, keyPath: \.view.view)
+
+    guard let renderedBody = renderer.body(for: view) else {
```

Comment:
I've only refactored `ViewDeferredToRenderer` for now. `SceneDeferredToRenderer` is kept as is to keep the PR small. And we didn't have problems with scenes (yet).